pkg/datastore/caveat.go

```diff
@@ -0,0 +1,26 @@
+package datastore
+
+import (
+	"errors"
+
+	core "github.com/authzed/spicedb/pkg/proto/core/v1"
+)
+
+// ErrCaveatNotFound is the error returned when a caveat is not found by either ID or name
+var ErrCaveatNotFound = errors.New("caveat not found")
```

Comment:
This should be a structured error, different for name vs ID, and include the name or ID in both the struct AND in the message

Reply:
addressed in https://github.com/authzed/spicedb/pull/807/commits/3ff40205f50abd9bfbc68c76ecd4ae7b3e1565b8